### ISSUE TITLE
Media failing to load on the site

### ISSUE THREAD

**Issue by rfraimow** [WGBH/AAPB2#1198]
I just hit a bunch of these when looking through the Voegeli collection -- went back to test this file on the AMS to make sure it existed, and it plays fine there, so the problems seems to be with the site

<img width="1159" alt="screen shot 2017-02-23 at 5 03 02 pm" src="https://cloud.githubusercontent.com/assets/5107596/23281174/39db6892-f9ea-11e6-9e88-a8de3e48f9b8.png">

**Comment by afred** [comment]
Checking the logs now, stand by.

**Comment by afred** [comment]
Here's what I'm getting in the logs when trying to access http://americanarchive.org/catalog/cpb-aacip_505-4m91834n3k (title: "I'll Be 91 Next Tuesday").
```
Processing by MediaController#show as */*
  Parameters: {"part"=>"1", "id"=>"cpb-aacip_505-4m91834n3k"}
Redirected to 
Completed 500 Internal Server Error in 377ms

ActionController::ActionControllerError (Cannot redirect to nil!):
  app/controllers/media_controller.rb:13:in `show'
  lib/middleware/redirect_middleware.rb:15:in `call'
```

This means that the bit of code responsible for looking up the Sony Ci URL for the media is not working... it's returning nothing, hence the `Cannot redirect to nil!` error message.

Can we confirm whether or not the media is in Ci? @rfraimow @caseyedavis12 @Muraszko-wgbh?

**Comment by afred** [comment]
The Sony Ci URL for the media is retrieved by making a call to `https://api.cimediacloud.com/assets/#{asset_id}/download`.

Hitting this api endpoint then returns a JSON object containing a property called `location`, which contains Sony Ci's temporary URL for the media.

I'm pretty sure that we have a value for `#{asset_id}` here (in this case `cpb-aacip_505-4m91834n3k` or some variation of that).

So my initial guesses at the problem are one of the following:
  1. The media is not currently in Sony Ci. This would cause the api call to return something that doesn't have a value for `location`, which would cause `nil` to be returned, and thus propagate the error we're seeing.
  1. If the media _IS_ currently on Sony Ci, then it could be one of the following...
    1. The `#{asset_id}` that we're using in our api call is not matching up with the record on Ci, and we're not getting a value back for `location`, which has the same result as the case above.
    1. Sony Ci's api has changed, and the value that is being returned from the api call is not the same as our current code is expecting. For instance, if they changed where we need to look for `location`, then that would also result in a `nil` value being returned.

So.. let's debug like this...
  1. First make sure the media is actually in Ci.
  1. If the media is in Ci, then let's make sure the IDs are lining up (i.e. verify that we can hit their api and get some kind of response other than 404 not found).
  1. If the media is up on Ci, and we're getting a response (other than 404 not found), then check the return value, and see if it has changed from what our code has been written to expect.

**Comment by afred** [comment]
cc @jasoncorum

**Comment by rfraimow** [comment]
I haven't been able to find any of the media for this collection in Ci by looking manually -- searching for cpb-aacip-505 turns up no items -- but that said, some of the items in the collection are playing from the site nonetheless, so it's perplexing either way.

**Comment by caseyedavis12** [comment]
@rfraimow: search for cpb-aacip_505 (underscore after aacip)

**Comment by rfraimow** [comment]
Ah, got it! OK, now I see most of them, and you're right, the asset in question is not currently in Ci.

**Comment by rfraimow** [comment]
The successful link in AMS shows src https://ci-buckets-assets-1umcaf2mqwhhg.s3.amazonaws.com/cifiles/758c855991b6478992d5771c18f97fc6/cpb-aacip_505-4m91834n3k_DV_2_21.aiff.mp3

**Comment by afred** [comment]
Can you get the media into Ci then?

**Comment by rfraimow** [comment]
I can, though it'll involve reindexing the affected assets on the AMS as well to provide the new Ci links. But if that's the best way to deal with this I'll move ahead with it. Any thoughts on how the media would have dropped out of Ci to begin with?

**Comment by afred** [comment]
> Any thoughts on how the media would have dropped out of Ci to begin with?

Nope.

**Comment by caseyedavis12** [comment]
@afred @rfraimow: when I search in Ci for cpb-aacip_505, I return 159 results. On the AAPB website, there are 214 digitized assets from station ID 505 (Donald Voegeli).


I also noticed that the file names for station ID 505 look like cpb-aacip_ and we usually name files like cpb-aacip-


Rebecca, can you provide a list of all Donald Voegeli assets that we know were uploaded to Sony Ci and a list of those that are no longer on Sony Ci?

**Comment by afred** [comment]
> I also noticed that the file names for station ID 505 look like cpb-aacip_ and we usually name files like `cpb-aacip-`

Remember that we convert the dash `-` to an underscore `_` when making AAPB URLs (for technical reasons).

However, the ID that is being used to hit the Sony Ci api comes from the pbcore xml at `/pbcoreIdentifier[@source="Sony Ci"]`

@caseyedavis12 @rfraimow Are these Sony Ci IDs the same as the uploaded filenames?

**Comment by rfraimow** [comment]
@afred sorry, not sure what you mean by this? the Ci IDs are never the same as the filenames

@caseyedavis12 I've attached the complete list of Voegeli assets and a list of the ones no longer to be found in Sony Ci
[voegli_ids.txt](https://github.com/WGBH/AAPB2/files/800156/voegli_ids.txt)
[voegeli_duplicate_proxies.xlsx](https://github.com/WGBH/AAPB2/files/800157/voegeli_duplicate_proxies.xlsx)

**Comment by afred** [comment]
@rfraimow Ah, ok. I'm not familiar with what the Sony Ci IDs look like (in code, everything is stored in variables). So if the Sony Ci IDs are completely different than the filenames, then it's less likely that the problem is related to having dashes vs. underscores in the IDs.

**Comment by rfraimow** [comment]
ah, gotcha. Yeah, the Ci IDs are always randomly generated by Ci after the file itself is uploaded.

**Comment by afred** [comment]
Ah.. right.. and then you have to go back and edit the PBCore to add the Sony Ci ID, right?

**Comment by afred** [comment]
How certain are we that the lost media files were uploaded to Ci?

**Comment by rfraimow** [comment]
Pretty positive -- I uploaded them myself, last May, and I have a list of all the Ci IDs that were generated then. Also, unless I'm missing something huge, I can't think of a way that the media files would be in the AMS if they didn't make it to Ci first, since the AMS also pulls media files from Ci.

At the time there were actually some duplicate Ci IDs uploaded from this data set, so I guess it's possible that someone got overzealous and deleted both the duplicate and the original from Ci? But that would only account for 29 of the 59 missing files.

**Comment by caseyedavis12** [comment]
Who all has delete access? Do we keep logs of everything that has been deleted?

**Comment by afred** [comment]
> At the time there were actually some duplicate Ci IDs uploaded from this data set, so I guess it's possible that someone got overzealous and deleted both the duplicate and the original from Ci?

But it's not possible to have 2 records in Sony Ci with the same ID, right? If you try to upload a duplicate, I assume it would create a new Ci ID for that record.

**Comment by rfraimow** [comment]
Yes, it would -- what you get is two Ci records with the same content, but different Ci IDs.

**Comment by afred** [comment]
Is it possible that dupes were uploaded to Ci, and then someone went in to delete the dupes, but deleted the one's whose Ci ID's were stored in the PBCore?

**Comment by rfraimow** [comment]
That's what I've been wondering, but it I looked in the trash folder and didn't find them -- also, it wouldn't account for the other 20-odd files.

**Comment by afred** [comment]
Ok.. so I'm getting timeouts when trying to hit the Sony Ci api from the AAPB webservers. The logs on production are a little hard to parse (since live traffic is flying by) but if you use Chrome's inspector, and click on the "Network" tab, you can see the failed request (highlighted in red).

To confirm locally, try this...
```
curl -L http://americanarchive.org/media/cpb-aacip_50-00000106?part=1 -e http://americanarchive.org;auto -o curl.out
curl: (7) Failed to connect to ci-buckets-assets-1umcaf2mqwhhg.s3.amazonaws.com port 443: Operation timed out
```

**Comment by rfraimow** [comment]
I will try this tomorrow when AWS is not in a total tailspin!

**Comment by Muraszko-wgbh** [comment]
@foo4thought @rfraimow 

So I decided to poke around this.  Looks like the originally uploaded file has a different name then the corrected one you uploaded recently.

Original (not playing on the public site)
```
cpb-aacip_505-4m91834n3k_DV_2_21.aiff.mp3
```

Recently updated 
```
cpb-aacip-505-4m91834n3k__DV_2_21.aiff.mp3
```

The recently updated one has a `-` and extra `_` in the filename.

The public AAPB site uses the filename to look up the thumbnails and media.  The AMS uses the Ci ID directly.  
So in the AMS, the pbcore has the Ci ID of `758c855991b6478992d5771c18f97fc6` which if you bring up that in the Sony Ci web app, [you'll see the misnamed audio file there](https://workspace.cimediacloud.com/ci/#/workspaces/051303c1c1d24da7988128e6d2f56aa9/folders/a1459cf719eb4a7cb2686663018be161?assets=758c855991b6478992d5771c18f97fc6).

**Comment by Muraszko-wgbh** [comment]
Also, should you ever need to verify a file with a Ci ID is actually in Ci, you can always use the ci.rb script you use to upload with the `--down Ci_ID` argument.
```
$ cd ~/AAPB2
$ ruby scripts/ci/ci.rb --down 758c855991b6478992d5771c18f97fc6
```

**Comment by rfraimow** [comment]
Yeah, I'm aware of the discrepancy in filenames between the original and the new proxies. Right now the public website is calling the original filename, so that didn't cause the problem; although the new structure (with the - and extra _) is expected by the AMS in order to assign Generation tags, but it doesn't have an impact on proxy playback to the best of my knowledge. 

Anyway, I tried using the --down argument to verify the original Ci IDs which the AMS is playing, and got the following error: 

scripts/ci/ci.rb:148:in `download': undefined method `[]' for nil:NilClass (NoMethodError)
	from scripts/ci/ci.rb:54:in `download'
	from scripts/ci/ci.rb:257:in `block in <main>'
	from scripts/ci/ci.rb:257:in `each'
	from scripts/ci/ci.rb:257:in `<main>'

**Comment by rfraimow** [comment]
OK -- Mike found the missing assets in the Stock Sales Ci workspace, which explains why the public site wasn't pulling them down. We are deleting them from Stock Sales and changing the records to point to the assets that I uploaded yesterday. Mystery solved, and I'm marking this as resolved.
